src/main/scala/firrtl/Utils.scala

```diff
@@ -12,11 +12,55 @@ import scala.collection.mutable.{StringBuilder, ArrayBuffer, LinkedHashMap, Hash
 import java.io.PrintWriter
 import logger.LazyLogging
 
-class FIRRTLException(str: String) extends Exception(str)
+object FIRRTLException {
+  def defaultMessage(message: String, cause: Throwable) = {
+    if (message != null) {
+      message
+    } else if (cause != null) {
+      cause.toString
+    } else {
+      null
+    }
+  }
+}
+class FIRRTLException(str: String, cause: Throwable = null)
+  extends RuntimeException(FIRRTLException.defaultMessage(str, cause), cause)
```

Comment:
What's the motivation for making FIRRTLException extend RuntimeException? Also, shouldn't we avoid null? I know this essentially wraps Java so perhaps it's unavoidable.